components/org.wso2.carbon.appmgt.gateway/src/main/java/org/wso2/carbon/appmgt/gateway/handlers/security/saml2/SAML2AuthenticationHandler.java

```diff
@@ -114,6 +152,8 @@ public void init(SynapseEnvironment synapseEnvironment) {
         try {
             authenticator = new OAuthAuthenticator();
             saml2Authenticator = new SAML2Authenticator();
+            webApp = new WebApp();
+            initJWTCapabilities();
         } catch (Exception e) {
             // Just throw it here - Synapse will handle it
             throw new SynapseException("Error while initializing SAML or OAuth authenticator");
```

Comment:
I think catching Exception is there to let the system initialize even partial errors. Seems OK, since it throws SynapseException. But better to pass the original exception for better understanding of the error when occurred. 
throw new SynapseException("Error while initializing SAML or OAuth authenticator", e);